src/python/pants/backend/java/target_types.py

```diff
@@ -149,5 +150,5 @@ def rules():
     return [
         *collect_rules(),
         *jvm_target_types.rules(),
-        *JavaFieldSet.jvm_rules(),
+        *jvm_rules(JavaFieldSet),
```

Comment:
Had to make `jvm_rules` a standalone function in order to break an import cycle.